From be08bb7a89d63b03476b4457c1dbc8265c4b241c Mon Sep 17 00:00:00 2001
From: Aaron Twining Myers <atm@apache.org>
Date: Wed, 17 Oct 2012 01:12:02 +0000
Subject: [PATCH 0601/1357] HDFS-3997. OfflineImageViewer incorrectly passes value of imageVersion when visiting IS_COMPRESSED element. Contributed by Mithun Radhakrishnan.

git-svn-id: https://svn.apache.org/repos/asf/hadoop/common/branches/branch-2@1399063 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit c98fd04db9ac9dfbaae7f5afc728d3f8d26afec7)
---
 .../offlineImageViewer/ImageLoaderCurrent.java     |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/offlineImageViewer/ImageLoaderCurrent.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/offlineImageViewer/ImageLoaderCurrent.java
index 2aade9e..391467b 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/offlineImageViewer/ImageLoaderCurrent.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/offlineImageViewer/ImageLoaderCurrent.java
@@ -167,7 +167,7 @@ class ImageLoaderCurrent implements ImageLoader {
 
       if (LayoutVersion.supports(Feature.FSIMAGE_COMPRESSION, imageVersion)) {
         boolean isCompressed = in.readBoolean();
-        v.visit(ImageElement.IS_COMPRESSED, imageVersion);
+        v.visit(ImageElement.IS_COMPRESSED, String.valueOf(isCompressed));
         if (isCompressed) {
           String codecClassName = Text.readString(in);
           v.visit(ImageElement.COMPRESS_CODEC, codecClassName);
-- 
1.7.0.4

